### PR TITLE
Don't use `new URL` for Wasm in `SINGLE_FILE` mode

### LINKED ISSUE #14309: Setting all three of SINGLE_FILE, EXPORT_ES6, and USE_ES6_IMPORT_META causes the contents of the .wasm file to be included twice in the generated file
This appears to be due to the code here: https://github.com/emscripten-core/emscripten/blob/3a1fcb84b58f529ec82b272adca4ffd30fdd4cd7/src/preamble.js#L749-L761

### PR DESCRIPTION
**Pull request by RReverser**
Fixes #14309.

### CODE REVIEW

File: src/preamble.js [hunk 1]
```diff
@@ -746,14 +746,14 @@ function instrumentWasmTableWithAbort() {
...
-#if EXPORT_ES6 && USE_ES6_IMPORT_META
+#if EXPORT_ES6 && USE_ES6_IMPORT_META && !SINGLE_FILE
 if (Module['locateFile']) {
 #endif
   var wasmBinaryFile = '{{{ WASM_BINARY_FILE }}}';
   if (!isDataURI(wasmBinaryFile)) {
     wasmBinaryFile = locateFile(wasmBinaryFile);
   }
-#if EXPORT_ES6 && USE_ES6_IMPORT_META
+#if EXPORT_ES6 && USE_ES6_IMPORT_META && !SINGLE_FILE
```
Comment (kripken) [inline src/preamble.js:756]:
```suggestion
#if EXPORT_ES6 && USE_ES6_IMPORT_META && !SINGLE_FILE // in single-file mode, repeating WASM_BINARY_FILE would emit the contents again
```

**Review by kripken — APPROVED** [review]

File: tests/test_other.py [hunk 2]
```diff
@@ -210,6 +210,17 @@ def test_emcc_output_worker_mjs(self, has_import_meta, args):
     src = read_file('subdir/hello_world.worker.js')
     self.assertContained('import("./hello_world.mjs")', src)
 
+  def test_emcc_output_worker_mjs_single_file(self):
+    os.mkdir('subdir')
```
Comment (sbc100) [inline tests/test_other.py:214]: This change doesn't seem to be related to threading or worker.js.. is it not enough to just test `.mjs` + `SINGLE_FILE=1`?   

Also do we need this `subdir` here?  We already have a test for that I think.

Comment (RReverser) [reply tests/test_other.py:214]: > This change doesn't seem to be related to threading or worker.js.. is it not enough to just test .mjs + SINGLE_FILE=1?

The change isn't but I think it's useful to have Workers in there too to verify that those still use the `import.meta.url` for Worker but not for Wasm, at least until #9796 is implemented.

Comment (sbc100) [reply tests/test_other.py:214]: I would at least like to see a more precise unit test that just tests the thing be fixed here `test_mjs_single_file`.     

Additionally testing the subdir handling and workers and pthreads seems out of scope for the fix.. and more of an integrated test than a unit test.     For me this kind of thing makes the test harder to maintain because it hard to see exactly what precisely is being tested (since many things are being tested at once).

Comment (RReverser) [reply tests/test_other.py:214]: It's testing that `import.meta.url` is used for Wasm but _not_ for Workers - it's too easy to assume those should be handled in the same way, and by checking both I want to make sure that someone doesn't look up blame and thinks "huh, he must have missed the Worker part" and proceeds to update it.

By copying both checks from above I'm ensuring that `import.meta` is handled correctly for both resources.

I agree the `subdir` bit is superfluous here though. Initially I was hoping to make this part of the parameterized test above, but due to different `data:` URL check it didn't quite work out :)

Comment (RReverser) [reply tests/test_other.py]: I've simplified test to check only the Wasm & Worker URLs - I think this is as minimal as it gets to prevent both issues mentioned above.

I did this via Github UI though, so let's wait for CI :)

**Review by sbc100 — APPROVED** [review]
